runtime/src/runtime/mod.rs

```diff
@@ -218,29 +224,53 @@ pub trait Runtime: Primitives + Verifier + RuntimePolicy {
     /// Returns the gas base fee (cost per unit) for the current epoch.
     fn base_fee(&self) -> TokenAmount;
 
+    /// The gas still available for computation
+    fn gas_available(&self) -> u64;
+
+    /// The current tipset's timestamp, as UNIX seconds
```

Comment:
Could you clarify whether "current" is the tipset that includes the message, or the subsequent one when the message is executed?

Reply:
Done!